src/olympia/activity/utils.py

```diff
@@ -0,0 +1,210 @@
+import base64
+import datetime
+import logging
+import quopri
+import re
+import urllib2
+from email import message_from_string
+from email.utils import parseaddr
+
+from django.conf import settings
+from django.template import Context, loader
+
+from email_reply_parser import EmailReplyParser
+
+from olympia import amo
+from olympia.access import acl
+from olympia.activity.models import ActivityLogToken
+from olympia.amo.helpers import absolutify
+from olympia.amo.urlresolvers import reverse
+from olympia.amo.utils import send_mail
+from olympia.devhub.models import ActivityLog
+
+log = logging.getLogger('z.amo.activity')
+
+# Prefix of the reply to address in devcomm emails.
+REPLY_TO_PREFIX = 'reviewreply+'
+
+
+class ActivityEmailParser(object):
+    """Utility to parse email replies."""
+    address_prefix = REPLY_TO_PREFIX
+
+    def __init__(self, email_text):
+        """Decode base64 email and turn it into a Django email object."""
+        try:
+            email_text = base64.standard_b64decode(
+                urllib2.unquote(email_text.rstrip()))
+        except TypeError:
+            # Corrupt or invalid base 64.
+            self.decode_error = True
+            log.info('Decoding error for ActivityEmailParser')
+            return
```

Comment:
I don't like the way this feels, either this should fail loudly and just raise the exception or wrap them in a custom exception so the outer layer can catch it better, especially since it's only being used by 
https://github.com/mozilla/addons-server/pull/3397/files#diff-68d7a6f301196292ff878b4e3665fa07R109

Something like

```python
class InvalidEmailText(ValueError):
    pass

class ActivityEmailParser(object):
    def __init__(self, email_text):
        try:
            email_text = base64.standard_b64decode(urllib2.unquote(email_text.rstrip()))
        except TypeError:
            log.exception('Decoding error for activity email reply parsing attempt')
            raise InvalidEmailText('Corrup or invalid base64 encoding')
```

So we can easily catch that in ``add_email_to_activity_log`` and handle accordingly.

I'm not a fan of setting class objects, thus using side-effects for error-handling.

Or as the Zen of Python would say
> Errors should never pass silently. And explicit is always better than implicit